akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -550,6 +620,7 @@ private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Mo
         case mv: MaterializedValueSource[_] ⇒
           val pub = new MaterializedValuePublisher
           materializedValuePublishers ::= pub
+          materializedValues.put(mv, ())
```

Comment:
This is not optional: without it the tests for accessing the materialized value fail. I don’t expect much impact on real-world benchmarks since this only affects usage of `FlowGraph.Builder.materializedValue`.